src/cr/cube/crunch_cube.py

```diff
@@ -713,6 +714,36 @@ def zscore(self, weighted=True, prune=False, hs_dims=None):
         res = [s.zscore(weighted, prune, hs_dims) for s in self.slices]
         return np.array(res) if self.ndim == 3 else res[0]
 
+    def pairwise_chisq(self, axis=0):
+        """Return square ndarray of pairwise Chi-square along axis.
+
+        Zscore is a measure of statistical signifficance of observed vs.
+        expected counts. It's only applicable to a 2D contingency tables.
+        For 3D cubes, the measures of separate slices are stacked together
+        and returned as the result.
+
+        :param weighted: Use weighted counts for zscores
+        :param prune: Prune based on unweighted counts
+        :param hs_dims: Include headers and subtotals (as NaN values)
+        :returns zscore: ndarray representing zscore measurements
+        """
+        res = [s.pairwise_chisq(axis=0) for s in self.slices]
+        return res[0]
+
+    def pairwise_pvals(self, axis=0):
+        """Return square ndarray of pairwise Chi-square along axis.
+
+        Square, symmetric matrix along *axis* of pairwise p-values for the
+        null hypothesis that col[i] = col[j] for each pair of columns.
+
+        :param axis: axis along which to perform comparison. Only columns (0)
+        implement currently.
+        :returns pvals: square, symmetric matrix of column-comparison p-values.
+        """
+        statistics = [s.pairwise_chisq(axis=0) for s in self.slices]
+        shapes = [s.get_shape() for s in self.slices]
+        return PairwisePvalues(statistics[0], shapes[0], axis=axis).values
```

Comment:
This class could take `self` as a parameter, and do all the work when needed, and not before...